jupyterlab/extensions/manager.py

```diff
@@ -0,0 +1,547 @@
+        enabled: [optional] Whether the package is enabled or not - default False
+        core: [optional] Whether the package is a core package or not - default False
+        latest_version: [optional] Latest available version - default ""
+        installed_version: [optional] Installed version - default ""
+        status: [optional] Package status - ["ok", "warning", "error"]; default "ok"
+        companion: [optional] Type of companion for the frontend extension - [None, "kernel", "server"]; default None
+        install: [optional] Extension package installation instructions - default None
+        installed: [optional] Whether the extension is currently installed - default None
+        is_allowed: [optional] Whether this extension is allowed or not - default True
+        author: [optional] Package author - default None
+        license: [optional] Package license - default None
+    """
+
+    name: str
+    description: str
+    url: str
+    pkg_type: str
+    companion: Optional[str] = None
+    core: bool = False
+    enabled: bool = False
+    install: Optional[dict] = None
+    installed: Optional[bool] = None
+    installed_version: str = ""
+    is_allowed: bool = True
+    latest_version: str = ""
+    status: str = "ok"
+    author: Optional[str] = None
+    license: Optional[str] = None
+
+
+@dataclass(frozen=True)
+class ActionResult:
+    """Action result
+
+    Attributes:
+        status: Action status - ["ok", "warning", "error"]
+        message: Action status explanation
+        needs_restart: Required action follow-up - Valid follow-up are "frontend", "kernel" and "server"
+    """
+
+    status: str  # FIXME
+    message: Optional[str] = None
+    needs_restart: List[str] = field(default_factory=list)
+
+
+@dataclass
+class ExtensionsOption:
+    """Extension manager options.
+
+    Attributes:
+        allowed_extensions_uris: A list of comma-separated URIs to get the allowed extensions list
+        blocked_extensions_uris: A list of comma-separated URIs to get the blocked extensions list
+        listings_refresh_seconds: The interval delay in seconds to refresh the lists
+        listings_tornado_options: The optional kwargs to use for the listings HTTP requests as described on https://www.tornadoweb.org/en/stable/httpclient.html#tornado.httpclient.HTTPRequest
+    """
+
+    allowed_extensions_uris: Set[str] = field(default_factory=set)
+    blocked_extensions_uris: Set[str] = field(default_factory=set)
+    listings_refresh_seconds: int = 60 * 60
+    listings_tornado_options: dict = field(default_factory=dict)
+
+
+@dataclass
+class ExtensionsCache:
+    """Extensions cache
+
+    Attributes:
+        cache: Extension list per page
+        last_page: Last available page result
+    """
+
+    cache: Dict[int, Optional[Dict[str, ExtensionPackage]]] = field(default_factory=dict)
+    last_page: int = 1
+
+
+class ExtensionsManager(LoggingConfigurable):
```

Comment:
Maybe this class could be renamed to `ExtensionManager` without the "s", since we usually refer to the "extension manager"?